### PR TITLE
Kate refactor

### PR DESCRIPTION
**Pull request by katemorris**
- Password Fields are now password fields
- All fields match now
- Default image is a full URL now, I give up there. 
- Inventory now cannot be negative and updated tests from that. 
- Fixed bug with merchants and admins seeing list of items. 
- Updated cart paths to be restful
- Updated tests around the merchant index page so that users could not perform those actions and admin user is now logged in for these tests. 
- Removed delete function on merchants
- Updates tests to stop some random failings.

### CODE REVIEW

File: spec/features/items/index_spec.rb [hunk 1]
```diff
@@ -132,4 +132,62 @@
       expect(page).to_not have_css('.item-statistics')
     end
   end
+
+  describe "When an admin visits the items index page" do
```
Comment (smcunning) [inline spec/features/items/index_spec.rb:136]: Nice job adding these tests!

**Review by smcunning — APPROVED** [review]
Nice job, Kate!

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -22,10 +22,10 @@
   <%= f.text_field :email, value: nil %>
 
   <%= f.label :password %>
-  <%= f.text_field :password, value: nil %>
+  <%= f.password_field :password, value: nil %>
 
   <%= f.label :password_confirmation %>
-  <%= f.text_field :password_confirmation, value: nil %>
+  <%= f.password_field :password_confirmation, value: nil %>
```
Comment (jdcarey128) [inline app/views/users/new.html.erb:29]: Good catch.

**Review by jdcarey128 — APPROVED** [review]

**Review by taphill — APPROVED** [review]
Great job Kate!

**Pull request merged**